genkit-tools/common/src/utils/utils.ts

```diff
@@ -14,59 +14,119 @@
  * limitations under the License.
  */
 
-import * as fs from 'fs';
+import axios from 'axios';
+import * as fs from 'fs/promises';
 import * as path from 'path';
-import { Runtime } from '../runner/types';
+import { Runtime } from '../manager/types';
 
-interface PackageJson {
-  main: string;
+/**
+ * Finds the project root by looking for a `package.json` file.
+ */
+export async function findProjectRoot(): Promise<string> {
+  let currentDir = process.cwd();
+  while (currentDir !== path.parse(currentDir).root) {
+    const packageJsonPath = path.join(currentDir, 'package.json');
```

Comment:
As a follow up, we might want to better handle CF3 case here. Package.json will be in the functions folder, but CWD might be at the root of firebase project folder, so this logic will fail to find the root.